Sources/Core/Permission.swift

```diff
@@ -41,5 +40,171 @@ enum Permission: Hashable {
   case userPosts
   case userTaggedPlaces
   case userVideos
-  case unknown(value: String)
+  case other(value: String)
+
+  // MARK: -
+
+  enum CodingKeys: String, CodingKey, CaseIterable {
+    case email
+    case groupsAccessMemberInfo
+    case publishToGroups
+    case userAgeRange
+    case userBirthday
+    case userEvents
+    case userFriends
+    case userGender
+    case userHometown
+    case userLikes
+    case userLink
+    case userLocation
+    case userMobilePhone
+    case userPhotos
+    case userPosts
+    case userTaggedPlaces
+    case userVideos
+    case other
+  }
+
+  // MARK: - Inits
+
+  init(from decoder: Decoder) throws {
+    let container = try decoder.singleValueContainer()
+    let val = try container.decode(String.self)
+    self.init(stringLiteral: val)
+  }
+
+  init(stringLiteral value: String) {
+    let keyVal = CodingKeys(rawValue: value) ?? CodingKeys(rawValue: value.camelCased())
+
+    switch keyVal {
+    case .email?:
+      self = .email
```

Comment:
Minor mappings like this make me think we should reconsider the swiftlint rule against single line case statements. This seems like a common enough and valid enough use case that we should be allowed to have these on one line. Even more so in the enum below where it's not even assignment but a direct one:one mapping.

Reply:
Lets address this later